Behave Reporting Support (#116)
testproject-io/python-sdk: README.rst

See `this page <https://docs.python.org/3/library/logging.html#logging-levels>`__ for a list of accepted logging levels and `look here <https://docs.python.org/3/howto/logging.html#changing-the-format-of-displayed-messages>`__ for more information on how to define a custom logging format.

Behave Support
--------------

> The SDK also supports automatic reporting of Behave features, scenarios and steps using the @behave_reporter annotation.

Reviewer: ```suggestion
The SDK also supports automatic reporting of Behave features, scenarios and steps using the @behave_reporter decorator.
```

> It will disable the reporting of driver command and automatic reporting of tests.

Reviewer: ```suggestion
It will disable the reporting of driver commands and automatic reporting of tests.
```

Instead, it will report:

* A test for every scenario in a feature file
* All steps in a scenario as steps in the corresponding test
* Steps are automatically marked as passed or failed, to create comprehensive living documentation from your
  specifications on TestProject Cloud.

> To enable Behave feature reporting, in your environment.py annotate the following methods:

Reviewer: ```suggestion
To enable Behave feature reporting, in your environment.py decorate one or more of the following methods:
```

> * method used to initialize your driver (usually, before_all or before_feature)

Reviewer: Please add a note that this is required in order to gain access to the driver from the behave context.

Author: Added note.

* after_step
* after_scenario

.. code-block:: python

    @behave_reporter
    def before_all(context):
        context.driver = webdriver.Chrome(projectname="Behave BDD")

    @behave_reporter
    def after_step(context, step):
        pass

    @behave_reporter
    def after_scenario(context, scenario):
        pass

By default, screenshots are taken only when step fail in your test, if you would like to change

> the behavior to always take a screenshot, pass the screenshot argument as True in your annotation.

Reviewer: ```suggestion
the behavior to always take a screenshot, pass the screenshot argument as `True` in your decorator.
```
